Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs

```diff
@@ -1106,11 +1121,29 @@ private void internalEventHandler(object sender, ManagerEvent e)
 					#endregion
 
 					case 62:
-						if (ConnectionState != null)
-						{
-							ConnectionState(this, (ConnectionStateEvent)e);
-							return;
-						}
+						if (Connect != null)

```

Comment:
Yes, but looking at the code, what you do is 
```if(Connect != null)```
So you're checking to see if the event has a handler. If it does, but the actual event being bubbled up is a reload event, then you'll still going to send it as 
```Connect(this, (ConnectEvent)e);```
because you've not checked what event is being passed to you in the switch statement, but instead you're checking if the event handler is null or not.
Do you follow me?